src/main/scala/org/opennetworkinsight/dns/DNSSuspiciousConnectsAnalysis.scala

```diff
@@ -0,0 +1,93 @@
+package org.opennetworkinsight.dns
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.opennetworkinsight.SuspiciousConnectsArgumentParser.SuspiciousConnectsConfig
+import org.opennetworkinsight.dns.DNSSchema._
+import org.opennetworkinsight.dns.model.DNSSuspiciousConnectsModel
+import org.opennetworkinsight.dns.sideinformation.DNSSideInformation
+import org.slf4j.Logger
+
+import org.opennetworkinsight.dns.model.DNSSuspiciousConnectsModel.ModelSchema
+import org.opennetworkinsight.dns.sideinformation.DNSSideInformation.SideInfoSchema
+
+/**
+  * The suspicious connections analysis of DNS log data develops a probabilistic model the DNS queries
+  * made by each client IP and flags
+  */
+
+object DNSSuspiciousConnectsAnalysis {
+
+  val inSchema = StructType(List(TimestampField, UnixTimestampField, FrameLengthField, ClientIPField,
+      QueryNameField, QueryClassField, QueryTypeField, QueryResponseCodeField))
+
+  val inColumns = inSchema.fieldNames.map(col)
+
+
+  assert(ModelSchema.fields.forall(inSchema.fields.contains(_)))
```

Comment:
should add a similar validation for the side field schema and outschema...

the idea is to catch a potential schema mismatch created by a customizing user as early as possible...

the biggest drawback with using dataframes is that schema checks occur only at runtime, not during typechecking... but I think that the easy management of columns outweighs that

still, we should make it so that any mucking about with the input/model/side/output schemas is validated before things get going